OK. We got the expected failure! Now let us try to fix it :smile: 

```
+ nbgrader -h
Traceback (most recent call last):
  File "/feedstock_root/build_artefacts/nbgrader_1500422916078/_t_env/bin/nbgrader", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/feedstock_root/build_artefacts/nbgrader_1500422916078/_t_env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3019, in <module>
    @_call_aside
  File "/feedstock_root/build_artefacts/nbgrader_1500422916078/_t_env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3003, in _call_aside
    f(*args, **kwargs)
  File "/feedstock_root/build_artefacts/nbgrader_1500422916078/_t_env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3032, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/feedstock_root/build_artefacts/nbgrader_1500422916078/_t_env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 655, in _build_master
    ws.require(__requires__)
  File "/feedstock_root/build_artefacts/nbgrader_1500422916078/_t_env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 963, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/feedstock_root/build_artefacts/nbgrader_1500422916078/_t_env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 849, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'jupyter' distribution was not found and is required by nbgrader
TESTS FAILED: nbgrader-v0.5.1-py27_0.tar.bz2
```

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -10,6 +10,14 @@ environment:
 
   matrix:
     - TARGET_ARCH: x86
+      CONDA_PY: 27
+      CONDA_INSTALL_LOCN: C:\\Miniconda
+
+    - TARGET_ARCH: x64
+      CONDA_PY: 27
+      CONDA_INSTALL_LOCN: C:\\Miniconda-x64
+
+    - TARGET_ARCH: x86
       CONDA_PY: 35
       CONDA_INSTALL_LOCN: C:\\Miniconda35
 

--- a/recipe/bld.bat
+++ b/recipe/bld.bat
@@ -1,3 +1,0 @@
-"%PYTHON%" setup.py install --single-version-externally-managed --record=record.txt
-CALL "%PREFIX%\Scripts\jupyter-nbextension" install --sys-prefix --overwrite --py nbgrader || EXIT /B 1
-IF %ERRORLEVEL% NEQ 0 EXIT /B %ERRORLEVEL%

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -1,2 +1,0 @@
-"${PYTHON}" setup.py install --single-version-externally-managed --record=record.txt
-"${PREFIX}/bin/jupyter-nbextension" install --sys-prefix --overwrite --py nbgrader

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,48 +1,32 @@
-{% set gh_org = "jupyter" %}
-{% set gh_repo = "nbgrader" %}
 {% set version = "v0.5.1" %}
-{% set sha256 = "971a37b58a8f1a26e7b17eacb091879b5f1646f5097b717626436a266a178d33" %}
 
 package:
-  name: {{ gh_repo|lower }}
+  name: nbgrader
   version: {{ version }}
 
 source:
-  fn: {{ gh_repo }}-{{ version }}.tar.gz
-  url: https://github.com/{{ gh_org }}/{{ gh_repo }}/archive/{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  url: https://github.com/jupyter/nbgrader/archive/{{ version }}.tar.gz
+  sha256: 971a37b58a8f1a26e7b17eacb091879b5f1646f5097b717626436a266a178d33
 
 build:
-  number: 0
-  skip: true  # [win and py27]
+  number: 1
+  script: python -m pip install --no-deps .
+  entry_points:
+    - nbgrader = nbgrader.apps.nbgraderapp:main
 
 requirements:
   build:
-    - jupyter
-    - jupyter_client
-    - jupyter_core
-    - nbconvert >=4.2
-    - nbformat
-    - notebook >=4.2
     - python
-    - python-dateutil
-    - requests
     - setuptools
-    - six
-    - sqlalchemy
-    - tornado
-    - traitlets
-    - jsonschema
-    - alembic
-
+    - pip
+    - jupyter
   run:
+    - python
+    - setuptools
     - jupyter
     - jupyter_client
     - jupyter_core
-    - nbconvert >=4.2
     - nbformat
-    - notebook >=4.2
-    - python
     - python-dateutil
     - requests
     - six
@@ -55,27 +39,24 @@ requirements:
 test:
   requires:
     - nodejs
-    - notebook >=4.2
     - pytest
-    - python
-    - requests
     - selenium
     - phantomjs
-
   source_files:
     - '*'
-
   imports:
     - nbgrader
   commands:
     - nbgrader -h
-  
+    - export USER=$(whoami)  # [not win]
+    - py.test -v -x -m "not nbextensions"
+
 about:
-  home: https://github.com/{{ gh_org }}/{{ gh_repo }}
+  home: https://github.com/jupyter/nbgrader
   license: BSD-3-Clause
   license_family: BSD
   license_file: COPYING.md
-  summary: A system for assigning and grading Jupyter notebooks.
+  summary: 'A system for assigning and grading Jupyter notebooks.'
 
 extra:
   recipe-maintainers:

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -49,7 +49,7 @@ test:
   commands:
     - nbgrader -h
     - export USER=$(whoami)  # [not win]
-    - py.test -v -x -m "not nbextensions"
+    - py.test -v -x -m "not nbextensions"  # [not (win and py2k)]
 
 about:
   home: https://github.com/jupyter/nbgrader

--- a/recipe/post-link.bat
+++ b/recipe/post-link.bat
@@ -1,5 +1,4 @@
 @echo off
-1>>"%PREFIX%\.messages.txt" 2>&1 (
-  "%PREFIX%\Scripts\jupyter-nbextension.exe" enable --sys-prefix --py nbgrader
-  "%PREFIX%\Scripts\jupyter-serverextension.exe" enable --sys-prefix --py nbgrader
-)
+
+"%PREFIX%\Scripts\jupyter-nbextension.exe" install nbgrader --py --sys-prefix >> "%PREFIX%\.messages.txt" 2>&1 && if errorlevel 1 exit 1
+"%PREFIX%\Scripts\jupyter-serverextension.exe" enable nbgrader --py --sys-prefix >> "%PREFIX%\.messages.txt" 2>&1 && if errorlevel 1 exit 1

--- a/recipe/post-link.sh
+++ b/recipe/post-link.sh
@@ -1,4 +1,2 @@
-{
-  "${PREFIX}/bin/jupyter-nbextension" enable --sys-prefix --py nbgrader
-  "${PREFIX}/bin/jupyter-serverextension" enable --sys-prefix --py nbgrader
-} >>"$PREFIX/.messages.txt" 2>&1
+"${PREFIX}/bin/jupyter-nbextension" install nbgrader --py --sys-prefix >> "${PREFIX}/.messages.txt" 2>&1
+"${PREFIX}/bin/jupyter-serverextension" enable nbgrader --py --sys-prefix >> "${PREFIX}/.messages.txt" 2>&1

--- a/recipe/pre-unlink.bat
+++ b/recipe/pre-unlink.bat
@@ -1,5 +1,4 @@
 @echo off
-1>>"%PREFIX%\.messages.txt" 2>&1 (
-  "%PREFIX%\Scripts\jupyter-nbextension" disable --sys-prefix --py nbgrader
-  "%PREFIX%\Scripts\jupyter-serverextension.exe" disable --sys-prefix --py nbgrader
-)
+
+"%PREFIX%\Scripts\jupyter-nbextension.exe" uninstall nbgrader --py --sys-prefix >> "%PREFIX%\.messages.txt" 2>&1 && if errorlevel 1 exit 1
+"%PREFIX%\Scripts\jupyter-serverextension.exe" disable nbgrader --py --sys-prefix >> "%PREFIX%\.messages.txt" 2>&1 && if errorlevel 1 exit 1

--- a/recipe/pre-unlink.sh
+++ b/recipe/pre-unlink.sh
@@ -1,4 +1,2 @@
-{
-  "${PREFIX}/bin/jupyter-nbextension" disable --sys-prefix --py nbgrader
-  "${PREFIX}/bin/jupyter-serverextension" disable --sys-prefix --py nbgrader
-} >>"$PREFIX/.messages.txt" 2>&1
+"${PREFIX}/bin/jupyter-nbextension" uninstall nbgrader --py --sys-prefix >> "${PREFIX}/.messages.txt" 2>&1
+"${PREFIX}/bin/jupyter-serverextension" disable nbgrader --py --sys-prefix >> "${PREFIX}/.messages.txt" 2>&1

--- a/recipe/run_test.bat
+++ b/recipe/run_test.bat
@@ -1,5 +1,0 @@
-CALL py.test -v -x -m "not nbextensions" || EXIT /B 1
-IF %ERRORLEVEL% NEQ 0 EXIT /B %ERRORLEVEL%
-
-rem CALL py.test -v -x -m "nbextensions" || EXIT /B 1
-rem IF %ERRORLEVEL% NEQ 0 EXIT /B %ERRORLEVEL%

--- a/recipe/run_test.sh
+++ b/recipe/run_test.sh
@@ -1,6 +1,0 @@
-# Make suer USER is set
-export USER=`whoami`
-
-# Call py.test directly as invoke causes CircleCI to bomb out, see #6
-py.test -v -x -m "not nbextensions"
-# py.test -v -x -m "nbextensions"

